EIPS/eip-4844.md

```diff
@@ -249,12 +249,17 @@ def validate_block(block: Block) -> None:
     for tx in block.transactions:
         ...
 
+        # destination must be not be empty
+        asset tx.to != None
         # the signer must be able to afford the transaction
         assert signer(tx).balance >= tx.gas * tx.max_fee_per_gas + get_total_data_gas(tx) * tx.max_fee_per_data_gas
-
         # ensure that the user was willing to at least pay the current data gasprice
         assert tx.max_fee_per_data_gas >= get_data_gasprice(parent(block).header)
 
+        # ensure the versioned hashes are non zero and valid
+        assert len(tx.blob_versioned_hashes) > 0
+        for versioned_hash in tx.blob_versioned_hashes:
+            assert versioned_hash[0] == BLOB_COMMITMENT_VERSION_KZG
```

Comment:
The change in PR #6583 implements it in a more accurate manner by first checking the transaction type.

Reply:
updated :+1: , however didn't fully import  #6583 changes which could be independently refactored there. let me know it thats fine or i can import it all over in this PR.